src/components/Guilds/Editor/index.tsx

```diff
@@ -0,0 +1,168 @@
+import styled, { css } from 'styled-components';
+import { transparentize } from 'polished';
+import { useEditor, EditorContent } from '@tiptap/react';
+import StarterKit from '@tiptap/starter-kit';
+import Focus from '@tiptap/extension-focus';
+import Highlight from '@tiptap/extension-highlight';
+import MenuBar from './MenuBar';
+import TurndownService from 'turndown';
+
+var turndownService = new TurndownService();
+
+const EditorWrap = styled.div`
+  background-color: #fff;
+  border: 1px solid ${({ theme }) => theme.colors.primary};
+  border-radius: 10px;
+  color: ${({ theme }) => theme.colors.primary};
+  display: flex;
+  flex-direction: column;
+  max-height: 26rem;
+}
+`;
+
+const Content = styled(EditorContent)`
+  flex: 1 1 auto;
+  overflow-x: hidden;
+  overflow-y: auto;
+  padding: 0 1rem;
+  -webkit-overflow-scrolling: touch;
+  ${css`
+    .ProseMirror {
+      > * + * {
+        margin-top: 0.75em;
+      }
+
+      outline: none;
+      min-height: 200px;
+
+      ul,
+      ol {
+        padding: 0 1rem;
+      }
+
+      h1,
+      h2,
+      h3,
+      h4,
+      h5,
+      h6 {
+        line-height: 1.1;
+      }
+
+      code {
+        background-color: ${({ theme }) =>
+          transparentize(0.8, theme.colors.muted)};
+        color: ${({ theme }) => theme.colors.muted};
+      }
+
+      pre {
+        background: ${({ theme }) => theme.colors.primary};
+        border-radius: 10px;
+        color: ${({ theme }) => theme.colors.background};
+        padding: 0.75rem 1rem;
+
+        code {
+          background: none;
+          color: inherit;
+          font-size: 0.8rem;
+          padding: 0;
+        }
+      }
+
+      mark {
+        background-color: #faf594;
+      }
+
+      img {
+        height: auto;
+        max-width: 100%;
+      }
+
+      hr {
+        margin: 1rem 0;
+      }
+
+      blockquote {
+        border-left: 2px solid
+          ${({ theme }) => transparentize(0.9, theme.colors.primary)};
+        padding-left: 1rem;
+      }
+
+      hr {
+        border: none;
+        border-top: 2px solid
+          ${({ theme }) => transparentize(0.8, theme.colors.primary)};
+        margin: 2rem 0;
+      }
+
+      ul[data-type='taskList'] {
+        list-style: none;
+        padding: 0;
+
+        li {
+          align-items: center;
+          display: flex;
+
+          > label {
+            flex: 0 0 auto;
+            margin-right: 0.5rem;
+            user-select: none;
+          }
+
+          > div {
+            flex: 1 1 auto;
+          }
+        }
+      }
+    }
+  `}
+`;
+
+const Editor = () => {
+  const initialJson = localStorage.getItem(
+    'guild-newProposal-description-json'

```

Comment:
Rather than doing `localStorage.getItem` and `localStorage.setItem` manually, I think we can use the `useLocalStorage` hook here? Just for reusability. I also tried to have a bit of a convention with the storage key names so far with a slash-separated format, but don't think that's very important anyway.

Reply:
Wow, forgot we had this, thanks!

Reply:
Will also use the ens convention